packages/v2/gui/src/sak/dokumenter/components/DocumentList.tsx

```diff
@@ -0,0 +1,258 @@
+import { Kommunikasjonsretning } from '@k9-sak-web/backend/k9sak/kodeverk/Kommunikasjonsretning.js';
+import { StarFillIcon } from '@navikt/aksel-icons';
+import { BodyShort, Label, Link, Select, Table, Tooltip } from '@navikt/ds-react';
+import { useQuery } from '@tanstack/react-query';
+import axios from 'axios';
+import { useState } from 'react';
+import DateTimeLabel from '../../../shared/dateTimeLabel/DateTimeLabel';
+import type { Document } from '../types/Document';
+import type { FagsakPerson } from '../types/FagsakPerson';
+import { type Kompletthet } from '../types/Kompletthetsperioder';
+import styles from './documentList.module.css';
+import arrowLeftPurpleImageUrl from './icons/arrow_left_purple.svg';
+import eksternLinkImageUrl from './icons/ekstern_link_pil_boks.svg';
+import internDokumentImageUrl from './icons/intern_dokument.svg';
+import mottaDokumentImageUrl from './icons/motta_dokument.svg';
+import sendDokumentImageUrl from './icons/send_dokument.svg';
+
+const headerTexts = ['Inn/ut', 'Dokument', 'Gjelder', 'Sendt/mottatt'];
+
+const alleBehandlinger = 'ALLE';
+
+const vedtaksdokumenter = ['INNVILGELSE', 'AVSLAG', 'FRITKS', 'ENDRING', 'MANUELL'];
+
+const inntektsmeldingBrevkode = '4936';
+
+const isVedtaksdokument = (document: Document) =>
+  vedtaksdokumenter.some(vedtaksdokument => vedtaksdokument === document.brevkode);
+
+const isTextMoreThan25char = (text?: string): boolean => !!text && text.length > 25;
+const trimText = (text: string): string => `${text?.substring(0, 24)}...`;
+
+const getDirectionImage = (document: Document): string => {
+  if (isVedtaksdokument(document)) {
+    return arrowLeftPurpleImageUrl;
+  }
+  if (document.kommunikasjonsretning === Kommunikasjonsretning.INN) {
+    return mottaDokumentImageUrl;
+  }
+  if (document.kommunikasjonsretning === Kommunikasjonsretning.UT) {
+    return sendDokumentImageUrl;
+  }
+  return internDokumentImageUrl;
+};
+const getDirectionText = (document: Document): string => {
+  if (document.kommunikasjonsretning === Kommunikasjonsretning.INN) {
+    return 'Inn';
+  }
+  if (document.kommunikasjonsretning === Kommunikasjonsretning.UT) {
+    return 'Ut';
+  }
+  return 'Intern';
+};
+
+const getModiaPath = (fødselsnummer?: string) => {
+  const { host } = window.location;
+  if (host === 'app-q1.adeo.no' || host === 'k9.dev.intern.nav.no') {
+    return `https://app-q1.adeo.no/modiapersonoversikt/person/${fødselsnummer}/meldinger/`;
+  }
+  if (host === 'app.adeo.no' || host === 'k9.intern.nav.no') {
+    return `https://app.adeo.no/modiapersonoversikt/person/${fødselsnummer}/meldinger/`;
+  }
+  return '#';
+};
+
+interface OwnProps {
+  documents: Document[];
+  behandlingId?: number;
+  fagsakPerson?: FagsakPerson;
+  saksnummer: number;
+  behandlingUuid: string;
+  sakstype: FagsakYtelsesType;
+}
+
+/**
+ * DocumentList
+ *
+ * Presentasjonskomponent. Viser dokumenter i en liste. Tar også inn en callback-funksjon som blir
```

Comment:
Er denne kommentar korrekt? Ser ingen callback funksjon i props.

Reply:
Takk, her gikk det for fort i svingene i kopieringen..